src/linker/configuration.py

```diff
@@ -100,6 +104,26 @@ def _get_steps(self) -> Tuple:
         if unknown_steps:
             logger.warning(
                 f"Unknown steps are included in the pipeline specification: {unknown_steps}.\n"
+                "These steps will be ignored.\n"
+                f"Supported steps: {STEP_ORDER}"
+            )
+        if not steps:
+            raise RuntimeError(
+                "No supported steps found in pipeline specification.\n"
+                f"Steps found in pipeline specification: {spec_steps}\n"
                 f"Supported steps: {STEP_ORDER}"
             )
+
         return steps
+
+    def _load_implementation_metadata(self) -> Dict[str, Path]:
```

Comment:
I think all this will go away once we have `Step` and `Implementation` defined as concrete objects, which you are doing in the next ticket, as implementations won't need to be tied directly to steps, whereas steps will have to define which steps they allow.